channelserver/src/main.rs

```diff
@@ -61,27 +62,28 @@ fn channel_route(req: &HttpRequest<session::WsChannelSessionState>) -> Result<Ht
         Some(id) => {
             // if the id is valid, but not present, treat it like a "None"
             if id.len() == 0 {
-                Uuid::new_v4()
+                channelid::ChannelID::default()
```

Comment:
Is `Default` the right abstraction here?  I'm not sure it makes sense to talk about "the default channel id" or similar, and I wonder if `ChannelID::new()` would be clearer about our intentions here.

Reply:
It seems to be kinda arbitrary. I've seen both `new()` and `default()` used to set initial values. I'm using `default()` here mostly because I don't think we ever want to intentionally create an unvalued channelid.

Reply:
It's pretty common to have a default for initial values like this -- e.g. you may have an Option<ChannelID> and find it handy to use unwrap_or_default ( clippy's mention of this expectation: https://github.com/rust-lang-nursery/rust-clippy/wiki#new_without_default )